dice.js

```diff
@@ -0,0 +1,294 @@
+  }
+
+  if (diceStillAvailable) {
+    // Three of a kind or more
+    for (i = 1; i < NUMBER_OF_DICE; i++) {
+      const dieCount = diceCounts[i];
+      if (dieCount >= 3) {
+        if (i < 4 || i === 5) {
+          // Twos, threes, fours
+          scoringOptions.push({
+            roll: `${dieCount} ${numberToWords.toWords(i + 1)}s`,
+            score: (i + 1) * 100 * (dieCount - 2),
+          });
+          removeScoredDice(i + 1, dieCount, chosenDice);
+        }
+      }
+    }
+
+    // Ones and Fives
+    switch (diceCounts[0]) {
+      case 0:
+        break;
+      case 1:
+        scoringOptions.push({ roll: "Single one", score: 100 });
+        removeScoredDice(1, diceCounts[0], chosenDice);
+        break;
+      case 2:
+        scoringOptions.push({ roll: `${diceCounts[0]} ones`, score: 200 });
+        removeScoredDice(1, diceCounts[0], chosenDice);
+        break;
+      default:
+        scoringOptions.push({
+          roll: `${diceCounts[0]} ones`,
+          score: 1000 * (diceCounts[0] - 2),
+        });
+        removeScoredDice(1, diceCounts[0], chosenDice);
+        break;
+    }
+
+    switch (diceCounts[4]) {
+      case 0:
+        break;
+      case 1:
+        scoringOptions.push({ roll: "Single five", score: 50 });
+        removeScoredDice(5, diceCounts[4], chosenDice);
+        break;
+      case 2:
+        scoringOptions.push({ roll: `${diceCounts[4]} fives`, score: 100 });
+        removeScoredDice(5, diceCounts[4], chosenDice);
+        break;
+      default:
+        scoringOptions.push({
+          roll: `${diceCounts[4]} fives`,
+          score: 500 * (diceCounts[4] - 2),
+        });
+        removeScoredDice(5, diceCounts[4], chosenDice);
+        break;
+    }
+  }
+
+  // Dice options that appear when there are no options within the chosen dice
+  if (scoringOptions.length === 0) {
+    let chosenDiceLength = 0;
+    let unavailableDice = 0;
+    for (let i = 0; i < NUMBER_OF_DICE; i++) {
+      chosenDiceLength += diceCounts[i];
+      if (!gameState.dice[i].available) unavailableDice++;
+    }
+    if (chosenDiceLength === NUMBER_OF_DICE) {
+      scoringOptions.push({ roll: "No scoring dice", score: 500 });
+      removeAllDice(removeAllDice);
+    } else if (chosenDiceLength + unavailableDice === 6) {
+      scoringOptions.push({ roll: "Zilch!", score: 0 });
+    } else {
+      scoringOptions.push("Choose some dice to see your options");
+    }
+  }
+
+  return scoringOptions;
+};
+
+// Changes to the next player's turn
+const nextPlayerTurn = (gameState) => {
+  // If the score limit has not been reached, it becomes the next player's turn
+  gameState.accumulatedPoints = 0;
+  gameState.potentialRollScore = 0;
+  gameState.currentPlayer =
+    gameState.currentPlayer < gameState.numberOfPlayers
+      ? gameState.currentPlayer + 1
+      : 1;
+  gameState.dice = rollDice();
+};
+
+/*
+while(nextPlayer.score < gameState.scoreLimit) {
+  clearTurnState();
+  nextPlayerTurn();
+}
+
+*/
```

Comment:
I would say don't remove this commented code because we will be using it later, but instead to specify it's for checking the match ending condition